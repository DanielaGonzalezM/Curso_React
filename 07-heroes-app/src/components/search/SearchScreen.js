import React from "react";
import { useForm } from "../hooks/useForm";

const SearchScreen = () => {
  const [formvalues, handleInputChange] = useForm({
    searchText: "",
  });

  const { searchText } = formvalues;
  const handleSearch = (e) => {
    e.preventDefault();
    console.log(searchText);
  };
  return (
    <>
      <h1>BÃºsqueda</h1>
      <hr />
      <div className="row">
        <div className="col-5">
          <h4>Buscar</h4>
          <hr />
          <form onSubmit={handleSearch}>
            <input
              type="text"
              placeholder="Buscar un heroe"
              className="form-control"
              name="searchText"
              autoComplete="off"
              value={searchText}
              onChange={handleInputChange}
            />
            <button type="submit" className="btn btn-outline-primary mt-1">
              Buscar...
            </button>
          </form>
        </div>
      </div>
    </>
  );
};

export default SearchScreen;
