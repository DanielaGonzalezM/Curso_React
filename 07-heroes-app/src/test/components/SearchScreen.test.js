import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import SearchScreen from "../../components/search/SearchScreen";

describe("Pruebas en SearchScreen", () => {
  test("should mostrarse correctamente con valores por defecto", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/search"]}>
        <SearchScreen />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".alert-info").text().trim()).toBe("Buscar un heroe");
  });
});
