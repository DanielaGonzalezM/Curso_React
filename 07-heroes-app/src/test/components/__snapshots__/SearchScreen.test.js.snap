// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en SearchScreen should mostrarse correctamente con valores por defecto 1`] = `
Array [
  <h1>
    BÃºsqueda
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Buscar
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Buscar un heroe"
          type="text"
          value=""
        />
        <button
          className="btn btn-outline-primary mt-1"
          type="submit"
        >
          Buscar...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Resultados
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Buscar un heroe
      </div>
    </div>
  </div>,
]
`;
